### PR TITLE
chore: set up cargo-dist

### PR DESCRIPTION
**Pull request by esteve**
This PR adds support for cargo-dist so we can distribute binaries that can be install via cargo-binstall

Fixes https://github.com/ros2-rust/cargo-ament-build/issues/9

### CODE REVIEW

**Comment by esteve** [comment]
@stelzo can I have you feedback on this? Thanks.

**Comment by stelzo** [comment]
Looks good and runs successfully - thank you for the quick patch.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+on:
+  pull_request:
+  push:
+    tags:
+      - '**[0-9]+.[0-9]+.[0-9]+*'
+
+jobs:
+  # Run 'cargo dist plan' (or host) to determine what tasks we need to do
+  plan:
+    runs-on: "ubuntu-20.04"
```
Comment (luca-della-vedova) [inline .github/workflows/release.yml:50]: Should this be `ubuntu-latest` rather than a version that will be EOL soon?

Comment (esteve) [reply .github/workflows/release.yml:50]: Good catch, thanks!

I don't know why `carg0-dist` defaults to `ubuntu-20.04` (see https://github.com/axodotdev/cargo-dist/blob/main/cargo-dist/src/backend/ci/github.rs#L500). I'll update this PR and submit a PR there to update the defaults.

Comment (esteve) [reply .github/workflows/release.yml:50]: PR for `cargo-dist` https://github.com/axodotdev/cargo-dist/pull/1403

**Comment by stelzo** [comment]
Changing the release.yml leads to a dirty state. I already tested it in [my main branch](https://github.com/stelzo/cargo-ament-build/actions/runs/10768357626/job/29857344466). You can see it running `cargo dist plan` locally. I would prefer staying with cargo-dist and switch to the new version when they update the runner.

**Comment by luca-della-vedova** [comment]
Ah I see, apologies not very familiar with `cargo-dist`, as long as there is some semi-automated way to keep the action up to date sounds good to me

**Comment by esteve** [comment]
@stelzo thanks for testing the changes, I've reverted the update.

**Pull request merged**